kde/qtcurve/Portfile

```diff
@@ -1,86 +1,241 @@
-	configure.args-delete -DCMAKE_BUILD_TYPE=Release
-	configure.args-append -DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo -DCMAKE_STRIP:FILEPATH=/bin/echo
-}
+subport ${name}-gtk2 {
+    PortGroup               cmake 1.1
+    long_description        The GTk2 version of the highly configurable QtCurve widget style for Qt
 
-variant qtonly description {Build only Qt style support, for installations without KDE} {}
-if { [variant_isset qtonly] } {
-	depends_lib-delete    port:kdelibs4
-#	configure.args-append -DQTC_QT_ONLY:BOOL=ON
-	configure.args-append -DQTC_QT4_ENABLE_KDE:BOOL=OFF
-} else {
+    post-patch {
+        # QtCurve-gtk2 has pure and optional runtime dependencies on KDE4 (and KF5)
+        reinplace "s|kde4-config|${prefix}/bin/kde4-config|g" \
+                            ${worksrcpath}/gtk2/style/qt_settings.cpp
+    }
+
+    # standard post-arg, where to find the primary CMakeLists.txt file.
+    default configure.post_args {../${worksrcdir}}
+    default configure.dir       {${workpath}/build}
+    default build.dir           {${workpath}/build}
+
+    variant LTO description {Build with Link-Time Optimisation (LTO)} {
+        configure.cflags-append     -flto
+        configure.cxxflags-append   -flto
+        configure.objcflags-append  -flto
+        configure.objcxxflags-append  -flto
+        configure.ldflags-append    -flto
+        # assume any compiler not clang will be gcc
+        if {![string match "*clang*" ${configure.compiler}]} {
+            configure.cflags-append         -fuse-linker-plugin -ffat-lto-objects
+            configure.cxxflags-append       -fuse-linker-plugin -ffat-lto-objects
+            configure.objcflags-append      -fuse-linker-plugin -ffat-lto-objects
+            configure.objcxxflags-append    -fuse-linker-plugin -ffat-lto-objects
+            configure.ldflags-append        -fuse-linker-plugin
+        }
+    }
+    depends_lib-append      port:gtk2
+    depends_run-append      port:perl5
+    configure.args-delete   -DQTC_ENABLE_X11:BOOL=OFF
+    configure.args-append   -DENABLE_GTK2:BOOL=ON -DQTC_ENABLE_X11:BOOL=ON \
+                            -DENABLE_QT4:BOOL=OFF -DENABLE_QT5:BOOL=OFF \
+                            -DQTC_QT4_ENABLE_KDE:BOOL=OFF -DQTC_QT5_ENABLE_KDE:BOOL=OFF \
+                            -DQTC_UTILSLIB_INFIX:STRING="-gtk2"
     post-destroot {
+        # impose Mac-like defaults
+        xinstall -m 644 ${filespath}/kdeglobals ${destroot}${prefix}/share/themes/QtCurve/gtk-2.0/kdeglobals
+    }
+}
+
+subport ${name}-extra {
+    long_description        Installs the Mac-specific default configuration plus additional \
+                            configuration presets and colour palettes for QtCurve
+    version                 ${qtc_version}
+    revision                2
+    epoch                   2
+    use_configure           no
+    installs_libs           no
+    supported_archs         noarch
+    depends_run-append      port:ciment-icons
```

Comment:
OTOH it might be better to introduce those two new ports separately in order to not overload this PR. It's already long enough. :-)